connector/src/test/java/com/google/cloud/hive/bigquery/connector/IntegrationTests.java

```diff
@@ -157,4 +161,148 @@ public void testUnsupportedTypes() {
+  }
+
+  @Test
+  public void testWhereClause() {
+    for (String engine : new String[] {"mr", "tez"}) {
+      for (String readDataFormat :
+          new String[] {HiveBigQueryConfig.ARROW /*, HiveBigQueryConfig.AVRO*/}) {
+        setUp();
+        whereClause(engine, readDataFormat);
+        tearDown();
+      }
+    }
+  }
+
+  // ---------------------------------------------------------------------------------------------------
+
+  /** Test the `SELECT` statement with explicit columns (i.e. not `SELECT *`) */
+  public void selectExplicitColumns(String engine, String readDataFormat) {
+    runBqQuery(BIGQUERY_TEST_TABLE_CREATE_QUERY);
+    initHive(engine, readDataFormat);
+    hive.execute(HIVE_TEST_TABLE_CREATE_QUERY);
+    // Insert data into BQ using the BQ SDK
+    runBqQuery(
+        String.format(
+            "INSERT `%s.%s` VALUES (123, 'hello'), (999, 'abcd')", DATASET, TEST_TABLE_NAME));
+    TableResult result =
+        runBqQuery(String.format("SELECT * FROM `%s.%s`", DATASET, TEST_TABLE_NAME));
+    // Make sure the initial data is there
+    assertEquals(2, result.getTotalRows());
+    // Read filtered data using Hive
+    // Try with both columns in order
+    List<Object[]> rows =
+        hive.executeStatement(
+            String.format("SELECT number, text FROM %s ORDER BY number", TEST_TABLE_NAME));
+    assertArrayEquals(
+        new Object[] {
+          new Object[] {123L, "hello"},
+          new Object[] {999L, "abcd"}
+        },
+        rows.toArray());
+    // Try in different order
+    rows =
+        hive.executeStatement(
+            String.format("SELECT text, number FROM %s ORDER BY number", TEST_TABLE_NAME));
+    assertArrayEquals(
+        new Object[] {
+          new Object[] {"hello", 123L},
+          new Object[] {"abcd", 999L}
+        },
+        rows.toArray());
+    // Try a single column
+    rows =
+        hive.executeStatement(
+            String.format("SELECT number FROM %s ORDER BY number", TEST_TABLE_NAME));
+    assertArrayEquals(new Object[] {new Object[] {123L}, new Object[] {999L}}, rows.toArray());
+    // Try another single column
+    rows =
+        hive.executeStatement(String.format("SELECT text FROM %s ORDER BY text", TEST_TABLE_NAME));
+    assertArrayEquals(new Object[] {new Object[] {"abcd"}, new Object[] {"hello"}}, rows.toArray());
+  }
+
+  @Test
+  public void testSelectExplicitColumns() {
+    for (String engine : new String[] {"mr", "tez"}) {
+      for (String readDataFormat :
+          new String[] {HiveBigQueryConfig.ARROW /*, HiveBigQueryConfig.AVRO*/}) {
+        setUp();
+        selectExplicitColumns(engine, readDataFormat);
+        tearDown();
+      }
+    }
+  }
+
+  // ---------------------------------------------------------------------------------------------------
+
+  /** Smoke test to make sure BigQuery accepts all different types of pushed predicates */
+  @Test
+  public void testWhereClauseAllTypes() {
```

Comment:
no asserts in this tests

Reply:
Yeah, this is more of a smoke test. Do you think we absolutely need an assertion?

Reply:
As a smoke test, it just checks that no exception gets thrown. I have other tests that check for validity of some types of filters. More tests could be added to verify the validity of all types of filters. But I thought a smoke test would be good enough for now.